WalletWasabi/Services/WalletService.cs

```diff
@@ -465,27 +451,51 @@ public List<SmartCoin> GetClusters(SmartCoin coin, List<SmartCoin> current, ILoo
 			return clusters;
 		}
 
-		private void ProcessBlock(Height height, Block block)
+		private void ProcessBlock(Height height, Block block, IEnumerable<int> filterByTxIds = null)
 		{
-			foreach (var tx in block.Transactions)
+			if (filterByTxIds is null)
+			{
+				var relevantIndicies = new List<int>();
+				for (int i = 0; i < block.Transactions.Count; i++)
+				{
+					Transaction tx = (Transaction)block.Transactions[i];
```

Comment:
This cast is redundant.
```suggestion
					Transaction tx = block.Transactions[i];
```

Reply:
ACK. Fixed.